### PR TITLE
Add tests for classes, add null checks

### PR DESCRIPTION
**Pull request by somratdutta**
## Overview
This PR introduces comprehensive tests for the `NatsJetStreamSinkWriter`, `NatsSinkWriter`, `NatsSourceReader` classes, add newline in EOF of `NatsSubjectSplitReaderTest` class.

## Key Changes
- Added NatsJetStreamSinkWriter  with thorough test coverage
- Added NatsSinkWriter  with thorough test coverage
- Added NatsSourceReader with thorough test coverage

## Test Coverage Changes

| Percentage       | Status                                   |
|------------------|------------------------------------------|
| 66.8% → 69.4%   |        |

### NatsSinkWriter
1. **Multi-Subject Publishing**
   - Tests publishing to multiple NATS subjects simultaneously
   - Validates message delivery and content across subjects
   
2. **Resource Management**
   - Tests proper cleanup on writer close
   - Validates write rejection after closure
 
### NatsJetStreamSinkWriter
1. **JetStream Message Publishing**
   - Tests successful message publishing to multiple JetStream subjects
   - Validates message delivery and content across subjects
   
2. **Resource Management**
   - Tests proper cleanup of resources on writer close
   - Validates rejection of write operations after closure

3. **Error Handling (TODO)**
   - Placeholder for testing JetStream publish failure scenarios

4. **String Representation**
   - Tests toString() method output format

### NatsSourceReader
1. **Basic Message Flow**
   - Validates core NATS to Flink message delivery
   - Tests subscription establishment and message deserialization
   
2. **Split State Management**
   - Tests multiple split handling
   - Validates state snapshot creation and persistence

3. **Message Availability Signaling**
   - Tests availability future behavior

4. **Error Handling**
   - Tests connection failure scenarios

5. **Source Event Handling**
   - Tests event processing behavior
   
6. **Split Completion Notification**
   - Ensures proper completion processing

### CODE REVIEW

File: src/main/java/io/synadia/flink/v0/source/reader/NatsSourceReader.java [hunk 1]
```diff
@@ -108,7 +108,9 @@ public void notifyNoMoreSplits() {
     @Override
     public void close() throws Exception {
         LOG.debug("{} | close", id);
-        connection.close();
+        if (connection != null) {
+            connection.close();
+        }
```
Comment (scottf) [inline src/main/java/io/synadia/flink/v0/source/reader/NatsSourceReader.java:113]: No. Please see my change in PR https://github.com/synadia-io/flink-connector-nats/pull/34

**Review by scottf — CHANGES_REQUESTED** [review]
I will re-review this after https://github.com/synadia-io/flink-connector-nats/pull/34 is accepted

File: src/main/java/io/synadia/flink/v0/source/reader/NatsSourceReader.java (hunk 1 above)
Comment (somratdnutanix) [reply src/main/java/io/synadia/flink/v0/source/reader/NatsSourceReader.java:113]: Sure, I will make the changes.

Comment (somratdutta) [reply src/main/java/io/synadia/flink/v0/source/reader/NatsSourceReader.java]: I have removed this change

**Review by scottf — APPROVED** [review]
LGTM
